src/ddmd/cppmangle.d

```diff
@@ -896,65 +898,57 @@ public:
         }
         else
         {
-            if (!substitute(t.sym))
+            Dsymbol s = t.sym;
+            Dsymbol p = s.toParent();
+            if (p && p.isTemplateInstance())
             {
-                cpp_mangle_name(t.sym, t.isConst());
+                if (substitute(p))
+                    return;
             }
-            if (t.isImmutable() || t.isShared())
+            if (!substitute(s))
             {
-                visit(cast(Type)t);
+                cpp_mangle_name(s, t.isConst());
             }
         }
         if (t.isConst())
-            store(t);
+             append(t);
```

Comment:
Looks like you added an extra space in your automated find/replacement of this function.

Reply:
indeed I did. oops!